### DISCUSSION TITLE
flexstruc so slow

### DISCUSSION THREAD

**Discussion by 1227419114** [DualSPHysics/DualSPHysics#212] [DUALSPHYSICS_V5.2]
What is a good solution to the problem of slow usage

**Comment by AhmedSalih3d** [comment]
Flexstruc is potentially slow, since it is the first implementation of new theory into the code base - most of the time completely new additions are not the fastest. The developers are working on improving it and through time it will most likely get faster and be able to used with mDBC as well.

Slow is also relative - if you share what you are simulating and simulation files you are using, it would be easier to figure out if something has been set up incorrectly.

Kind regards

**Comment by 1227419114** [comment]
"Hello, I apologize for the late response. Currently, I have an issue where the simulated amplitude of vibration in a two-degree-of-freedom VIV simulation is larger than the actual value. Is there a solution?"

Below are my VIV video and XLM file













At 2024-08-19 21:56:39, "AhmedSalih3d" ***@***.***> wrote:

Flexstruc is potentially slow, since it is the first implementation of new theory into the code base - most of the time completely new additions are not the fastest. The developers are working on improving it and through time it will most likely get faster and be able to used with mDBC as well.

Slow is also relative - if you share what you are simulating and simulation files you are using, it would be easier to figure out if something has been set up incorrectly.

Kind regards

**Comment by YiOuO** [comment]
Hi, the time step size for strucutre particles depends on the sound of speed in solid, which is always very large
